WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java

```diff
@@ -0,0 +1,337 @@
+package org.wordpress.android.ui.prefs;
+
+import android.os.Bundle;
+import android.preference.EditTextPreference;
+import android.preference.ListPreference;
+import android.preference.Preference;
+import android.preference.PreferenceFragment;
+
+import com.android.volley.VolleyError;
+import com.wordpress.rest.RestRequest;
+
+import org.json.JSONObject;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.models.Blog;
+import org.wordpress.android.networking.RestClientUtils;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.ToastUtils;
+
+import java.util.HashMap;
+import java.util.Locale;
+
+/**
+ * Handles changes to WordPress site settings. Syncs with host automatically when user leaves.
+ */
+
+public class SiteSettingsFragment extends PreferenceFragment
+        implements Preference.OnPreferenceChangeListener {
+    private HashMap<String, String> mLanguageCodes = new HashMap<>();
+    private Blog mBlog;
+    private EditTextPreference mTitlePreference;
+    private EditTextPreference mTaglinePreference;
+    private EditTextPreference mAddressPreference;
+    private ListPreference mLanguagePreference;
+    private ListPreference mPrivacyPreference;
+
+    // Most recent remote site data. Current local data is used if remote data cannot be fetched.
+    private String mRemoteTitle;
+    private String mRemoteTagline;
+    private String mRemoteAddress;
+    private int mRemotePrivacy;
+    private String mRemoteLanguage;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // make sure we have local site data
+        mBlog = WordPress.getBlog(
+                getArguments().getInt(BlogPreferencesActivity.ARG_LOCAL_BLOG_ID, -1));
+        if (mBlog == null) return;
+
+        // inflate Site Settings preferences from XML
+        addPreferencesFromResource(R.xml.site_settings);
+
+        // set preference references, add change listeners, and setup various entries and values
+        initPreferences();
+
+        // fetch remote site data
+        fetchRemoteData();
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+
+        // Assume user wanted changes propagated when they leave
+        applyChanges();
```

Comment:
It would be more bandwidth-friendly to only call `applyChanges()` when the user has actually made changes.